### PR TITLE
[vehicle cmd gate] porting vehicle cmd gate

### PR DESCRIPTION
**Pull request by Sugatyon**
## Description

msgs below are not ported to .auto msgs
- autoware_external_api_msgs
- autoware_debug_msgs
- gate_mode

To do
- add readme

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp [hunk 1]
```diff
@@ -381,69 +420,62 @@ void VehicleCmdGate::publishControlCommands(const Commands & commands)
...
 
-  // Shift
-  autoware_vehicle_msgs::msg::ShiftStamped shift;
-  shift.header.stamp = stamp;
-  shift.header.frame_id = "base_link";
-  shift.shift.data = autoware_vehicle_msgs::msg::Shift::NONE;
+  // gear
+  autoware_auto_vehicle_msgs::msg::GearCommand gear;
+  gear.stamp = stamp;
+  // default value is 0
```
Comment (takayuki5168) [inline control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp:460]: What do we assign to gear command here? When emergency, the ego vehicle tries to deccel and stop, so DRIVE?

File: control/vehicle_cmd_gate/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Vehicle Cmd Decider
```
Comment (takayuki5168) [inline control/vehicle_cmd_gate/README.md:1]:
```suggestion
# vehicle_cmd_gate
```

File: control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp (hunk 1 above)
Comment (Sugatyon) [reply control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp:460]: NONE is assigned as a default, so i didn't assign anything. 
If DRIVE should be assigned, then autoware_vehicle_msgs::msg::Shift::DRIVE was assigned.

Comment (takayuki5168) [reply control/vehicle_cmd_gate/src/vehicle_cmd_gate.cpp:460]: NONE(0) seems better, okay!

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Pull request merged**